### PR TITLE
USER_AUTHENTICATION_MQCSP switch Added as config param

### PR DESCRIPTION
**Pull request by mtghunt**

### CODE REVIEW

File: src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java [hunk 1]
```diff
@@ -213,13 +217,17 @@
...
+                      CONFIG_DISPLAY_MQ_CSP);
 
         config.define(CONFIG_NAME_MQ_PASSWORD, Type.PASSWORD, null, Importance.MEDIUM,
                       CONFIG_DOCUMENTATION_MQ_PASSWORD, CONFIG_GROUP_MQ, 8, Width.MEDIUM,
                       CONFIG_DISPLAY_MQ_PASSWORD);
 
+        config.define(CONFIG_NAME_MQ_CSP, Type.STRING, true, Importance.MEDIUM,
+                CONFIG_DOCUMENTATION_MQ_USER_NAME, CONFIG_GROUP_MQ, 7, Width.MEDIUM,
+                CONFIG_DISPLAY_MQ_USER_NAME);
+
```
Comment (flojon) [inline src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java]: Looks like CONFIG_DOCUMENTATION_MQ_* and CONFIG_DISPLAY_MQ_* variables has been mixed up on lines 220-221 + 228-229

File: src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java [hunk 2]
```diff
@@ -213,13 +217,17 @@
...
-                      CONFIG_DISPLAY_MQ_USER_NAME);
+                      CONFIG_DOCUMENTATION_MQ_CSP, CONFIG_GROUP_MQ, 7, Width.MEDIUM,
+                      CONFIG_DISPLAY_MQ_CSP);
 
         config.define(CONFIG_NAME_MQ_PASSWORD, Type.PASSWORD, null, Importance.MEDIUM,
                       CONFIG_DOCUMENTATION_MQ_PASSWORD, CONFIG_GROUP_MQ, 8, Width.MEDIUM,
                       CONFIG_DISPLAY_MQ_PASSWORD);
 
+        config.define(CONFIG_NAME_MQ_CSP, Type.STRING, true, Importance.MEDIUM,
+                CONFIG_DOCUMENTATION_MQ_USER_NAME, CONFIG_GROUP_MQ, 7, Width.MEDIUM,
```
Comment (flojon) [inline src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java]: Should 7 here be unique? So maybe 10?

File: src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java [hunk 3]
```diff
@@ -61,10 +61,14 @@
     public static final String CONFIG_DOCUMENTATION_MQ_USER_NAME = "The user name for authenticating with the queue manager.";
     public static final String CONFIG_DISPLAY_MQ_USER_NAME = "User name";
 
-    public static final String CONFIG_NAME_MQ_PASSWORD = "mq.password"; 
+    public static final String CONFIG_NAME_MQ_PASSWORD = "mq.password";
     public static final String CONFIG_DOCUMENTATION_MQ_PASSWORD = "The password for authenticating with the queue manager.";
     public static final String CONFIG_DISPLAY_MQ_PASSWORD = "Password";
 
+    public static final String CONFIG_NAME_MQ_CSP = "mq.csp";
+    public static final String CONFIG_DOCUMENTATION_MQ_CSP = "useMQCSPauthentication.";
```
Comment (AndrewJSchofield) [inline src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java]: Should be a grammatical sentence.

File: src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java [hunk 4]
```diff
@@ -61,10 +61,14 @@
...
+    public static final String CONFIG_DISPLAY_MQ_CSP = "useMQCSPauthentication";
```
Comment (AndrewJSchofield) [inline src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java]: This is intended to be presented in a UI alongside the other options. This one doesn't follow the existing style.

File: src/main/java/com/ibm/eventstreams/connect/mqsource/JMSReader.java [hunk 5]
```diff
@@ -97,26 +97,27 @@ public void configure(Map<String, String> props) {
...
-            }               
+            }
         }
 
         try {
             mqConnFactory = new MQConnectionFactory();
             mqConnFactory.setTransportType(transportType);
             mqConnFactory.setQueueManager(queueManager);
-            mqConnFactory.setBooleanProperty(WMQConstants.USER_AUTHENTICATION_MQCSP, true);
+            mqConnFactory.setBooleanProperty(WMQConstants.USER_AUTHENTICATION_MQCSP, Boolean.parseBoolean(csp));
```
Comment (AndrewJSchofield) [inline src/main/java/com/ibm/eventstreams/connect/mqsource/JMSReader.java]: This needs to default to 'true'. The new code defaults to 'false'.

File: src/main/java/com/ibm/eventstreams/connect/mqsource/JMSReader.java [hunk 6]
```diff
@@ -351,7 +352,8 @@ private boolean connectInternal() {
                 Thread.sleep(reconnectDelayMillis);
             }
             catch (InterruptedException ie) {
-                ;
+                log.error("Thread.sleep failed in connectInternal()", jmse);
```
Comment (AndrewJSchofield) [inline src/main/java/com/ibm/eventstreams/connect/mqsource/JMSReader.java]: This PR should not include other stuff too. Please restrict it to configuration of a CSP option only.

File: src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java (hunk 1 above)
Comment (mtghunt) [reply src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java]: Done

File: src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java (hunk 3 above)
Comment (mtghunt) [reply src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java]: fixed

File: src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java (hunk 2 above)
Comment (mtghunt) [reply src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java]: fixed

File: src/main/java/com/ibm/eventstreams/connect/mqsource/JMSReader.java (hunk 5 above)
Comment (mtghunt) [reply src/main/java/com/ibm/eventstreams/connect/mqsource/JMSReader.java]: done

File: src/main/java/com/ibm/eventstreams/connect/mqsource/JMSReader.java (hunk 6 above)
Comment (mtghunt) [reply src/main/java/com/ibm/eventstreams/connect/mqsource/JMSReader.java]: returned ";"  :)

File: src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java (hunk 4 above)
Comment (mtghunt) [reply src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java]: done

**Comment by AndrewJSchofield** [comment]
Still a few nits. The config options are not quite right. The option needs adding to the README. I know what you want and am happy to put it in the connector (and the sink too). Let me know whether you want to continue with me reviewing the PR until it's perfect or just implement the feature for you. Either way is good for me.

**Comment by AndrewJSchofield** [comment]
This PR was inactive for a few weeks so I have implemented in another PR.

**Pull request closed without merging**